plugins/colorsource/export.js

```diff
@@ -0,0 +1,252 @@
+        }
+
+        // remove null values and empty arrays
+        Object.entries(channelJson).forEach(([key, value]) => {
+          if (value === null || (Array.isArray(value) && value.length === 0)) {
+            delete channelJson[key];
+          }
+        });
+
+        fixtureJson.parameters.push(channelJson);
+      });
+
+      fixtureJson.colortable = getColorTable(fixtureJson.parameters);
+
+      // remove null values and empty arrays
+      Object.entries(fixtureJson).forEach(([key, value]) => {
+        if (value === null || (Array.isArray(value) && value.length === 0)) {
+          delete fixtureJson[key];
+        }
+      });
+
+      exportJson.personalities.push(fixtureJson);
+
+
+      /**
+       * @param {!AbstractChannel} channel The OFL channel of which the ColorSource channel type should be returned.
+       * @returns {number} One of ColorSource's channel types as positive integer.
+       */
+      function getCSChannelType(channel) {
+        if (channel.type === `NoFunction`) {
+          return 0;
+        }
+
+        if (channel.type === `Single Color` || [`Hue`, `Saturation`].includes(channel.name)) {
+          return 5;
+        }
+
+        if (channel.type === `Intensity`) {
+          return 1;
+        }
+
+        if ([`Pan`, `Tilt`].includes(channel.type)) {
+          return 2;
+        }
+
+        if (channel.type === `Color Temperature`) {
+          return 3;
+        }
+
+        return 4;
+      }
+
+      /**
+       * Adds detailed information to given channel JSON that only a CoarseChannel can provide.
+       * @param {object} channelJson The ColorSource channel JSON to which the data is added.
+       * @param {CoarseChannel} channel The channel whose information should be used.
+       * @param {number} channelIndex The position of the channel in the current mode, starting from zero.
+       */
+      function addChannelDetails(channelJson, channel, channelIndex) {
+        channelJson.fadeWithIntensity = channel.type === `ColorIntensity` && hasIntensity;
+
+        const fineChannel16bit = channel.fineChannels[0];
+        const fineChannelIndex = mode.getChannelIndex(fineChannel16bit || {}, `default`);
+        if (fineChannelIndex !== -1) {
+          channelJson.fine = fineChannelIndex;
+          channelJson.size = 16;
+        }
+
+        const channelResolution = channelJson.size / 8;
+        channelJson.highlight = scaleDmxValue(channel.getHighlightValueWithResolution(channelResolution), channelResolution, CoarseChannel.RESOLUTION_16BIT);
+        channelJson.home = scaleDmxValue(channel.getDefaultValueWithResolution(channelResolution), channelResolution, CoarseChannel.RESOLUTION_16BIT);
+        channelJson.invert = channel.isInverted;
+        channelJson.snap = !channel.canCrossfade;
+
+        channel.capabilities.forEach(cap => {
+          const dmxRange = cap.getDmxRangeWithResolution(CoarseChannel.RESOLUTION_8BIT);
```

Comment:
Are DMX range and menu click values really 8bit, while highlight and default value are always 16bit?

Reply:
Yes indeed, they really are.